### ISSUE TITLE
Y24-275 - As PM (Adrianne L) I would like to 3 controls to be introduced when creating the RVI Cherrypick plate so that we can ensure the quality of the pipeline.

### ISSUE THREAD

**Issue by TWJW-SANGER** [sanger/limber#1864]
**User story**
As PM (Adrianne L) I would like to 3 controls to be introduced when creating the RVI Cherrypick plate so that we can ensure the quality of the pipeline.

**Who are the primary contacts for this story**
AL ST

**Who is the nominated tester for UAT**
TBD

**Acceptance criteria**

Are there specific controls (positive or negative) that need be added?
Will this be in random location or is there a fixed location/s ?

To be considered successful the solution must allow:
- [ ] add a list of acceptance criteria here
- [ ] ...

**Additional context**

**Comment by KatyTaylor** [comment]
Further email from Ya-Lin:

> After liaising with SSR, we believe the easiest way to achieve our goal is adding a template to the “Control Plate” menu for CP submission under Automatic layout (see pic attached) to automatically assign controls to H10, H11, and H12 wells.
 >
> Please kindly let us know whether this sounds like a plan and if yes what the possible timeline of implementing this change would be.

I think this is correct, that all we need to do is to add a new entry into the Control Plate drop down. It looks like it automatically adds the controls onto each destination plate, and randomises the location.

**Comment by KatyTaylor** [comment]
Testing SS cherrypicking using:
* a 96-well stock plate
* Plate Template 'H12 empty'
* Control Plate 'Heron A1+ H12-'

Typical layout is shown below. 
* Control locations are 'randomised' on the first (full) destination plate
* They are added at the end of the second (partially filled) plate
* They can be in the same row
* In 10 submissions, they did not appear in the same column (need to check in code if this is enforced)
* H12 was left empty as per template

![Screenshot 2024-08-23 at 15 25 43](https://github.com/user-attachments/assets/f52b5f04-92a2-4ff5-8600-7072f6e24a95)

**Comment by KatyTaylor** [comment]
Looking into whether we can use the Plate Template dropdown to put controls in fixed positions.

**TL;DR**
Looks to me like Plate Templates can only be used to leave wells empty - not to populate wells with specific samples.

**Further information**
See `initialize_already_occupied_wells_from` and `add_any_wells_from_template_or_partial` in `CherrypickTask::PickTarget::Base` (pick_target.rb). 

A Plate Template is a Labware with sti_type PlateTemplate. Most existing Plate Templates in the UAT environment have a small number of wells, which contain no aliquots or samples e.g. [H12 empty](https://uat.sequencescape.psd.sanger.ac.uk/labware/3613830) has one well, H12, which does not contain an aliquot or sample. There is at least one Plate Template in the UAT environment which has samples in it ([QcReportersV1](https://uat.sequencescape.psd.sanger.ac.uk/labware/9978093)) - not sure what this is used for. It does not transfer the samples across when cherrypicking, it just leaves them empty.

**Comment by KatyTaylor** [comment]
Options for meeting the RVI cherrypicking requirement:
1. Modify existing control randomisation behaviour to disallow multiple controls in the same column (would affect other users of cherrypicking, but that might not be a problem).
2. Modify control randomisation to include column constraint - but as an option, so as not to change behaviour for other users.
3. Add a way to add controls into fixed locations.

**Comment by KatyTaylor** [comment]
Answers to above questions from Ya-Lin (email):

> Both ways you’ve proposed sound good to me. There’s no preference from the project requirement. Which one PSD can get done quicker would be our choice I believe 😊

> Regarding the partial plate, as currently we only put full plates into pipeline, this shouldn’t cause any concern. When partial plates are to be picked, we don’t usually add controls yet. Instead, we’ll set up another cherrypicking to combine partial plates into a full plate and add the controls at this step before putting the plate into pipeline.

**Comment by KatyTaylor** [comment]
Output - https://github.com/sanger/sequencescape/issues/4333
